### PR TITLE
StandaloneMmPkg/MmIpl : Check if  MM FV HOB was built

### PR DESCRIPTION
**Pull request by hongbin123**
- [ ] Breaking change?
  - **Breaking change** - Does this PR cause a break in build or boot behavior?
  - Examples: Does it add a new library class or move a module to a different repo.
- [ ] Impacts security?
  - **Security** - Does this PR have a direct security impact?
  - Examples: Crypto algorithm change or buffer overflow fix.
- [ ] Includes tests?
  - **Tests** - Does this PR include any explicit test code?
  - Examples: Unit tests or integration tests.

### CODE REVIEW

File: StandaloneMmPkg/Drivers/StandaloneMmIplPei/MmFoundationHob.c [hunk 1]
```diff
@@ -803,6 +803,49 @@ GetRemainingHobSize (
...
+  if No, continue to build MM Core FV HOB
+
+  @param[in]      PlatformHobList     Platform HOB list.
+  @param[in]      PlatformHobSize     Platform HOB size.
+
+  @retval TRUE    Skip building MM Core FV HOB.
+          FALSE   Continue to build MM Core FV HOB.
+**/
+BOOLEAN
+IsMmCoreFvHobBuilt (
```
Comment (jiaxinwu) [inline StandaloneMmPkg/Drivers/StandaloneMmIplPei/MmFoundationHob.c:820]: why not merge this PR into the https://github.com/tianocore/edk2/pull/6547 as one of commit?

File: StandaloneMmPkg/Drivers/StandaloneMmIplPei/MmFoundationHob.c [hunk 2]
```diff
@@ -894,9 +937,11 @@ CreateMmFoundationHobList (
   //
   // BFV address for StandaloneMm Core
   //
-  HobLength = GetRemainingHobSize (*FoundationHobSize, UsedSize);
-  MmIplBuildFvHob (FoundationHobList + UsedSize, &HobLength, MmFvBase, MmFvSize);
-  UsedSize += HobLength;
+  if (IsMmCoreFvHobBuilt (PlatformHobList, PlatformHobSize) == FALSE) {
```
Comment (jiaxinwu) [inline StandaloneMmPkg/Drivers/StandaloneMmIplPei/MmFoundationHob.c:940]: if (!IsPlatformBuildFvHob (PlatformHobList, PlatformHobSize)) {}

File: StandaloneMmPkg/Drivers/StandaloneMmIplPei/MmFoundationHob.c (hunk 1 above)
[the remark this answers is not in the document]
Comment (jiaxinwu) [reply StandaloneMmPkg/Drivers/StandaloneMmIplPei/MmFoundationHob.c:820]: can be named to IsPlatformBuildFvHob

[the remark this answers is not in the document]
Comment (hongbin123) [reply StandaloneMmPkg/Drivers/StandaloneMmIplPei/MmFoundationHob.c]: updated

File: StandaloneMmPkg/Drivers/StandaloneMmIplPei/MmFoundationHob.c (hunk 2 above)
Comment (hongbin123) [reply StandaloneMmPkg/Drivers/StandaloneMmIplPei/MmFoundationHob.c]: updated

File: StandaloneMmPkg/Drivers/StandaloneMmIplPei/MmFoundationHob.c [hunk 3]
```diff
@@ -803,6 +803,49 @@ GetRemainingHobSize (
...
+  @param[in]      PlatformHobList     Platform HOB list.
+  @param[in]      PlatformHobSize     Platform HOB size.
+
+  @retval TRUE    Skip building MM Core FV HOB.
+          FALSE   Continue to build MM Core FV HOB.
+**/
+BOOLEAN
+IsMmPlatformFvHobCreated (
+  IN UINT8  *PlatformHobList,
+  IN UINTN  PlatformHobSize
```
Comment (niruiyu) [inline StandaloneMmPkg/Drivers/StandaloneMmIplPei/MmFoundationHob.c:822]: Change to "IsFvHobExist (HobList, HobSize)"

File: StandaloneMmPkg/Drivers/StandaloneMmIplPei/MmFoundationHob.c [hunk 4]
```diff
@@ -894,9 +937,11 @@ CreateMmFoundationHobList (
   //
   // BFV address for StandaloneMm Core
   //
-  HobLength = GetRemainingHobSize (*FoundationHobSize, UsedSize);
-  MmIplBuildFvHob (FoundationHobList + UsedSize, &HobLength, MmFvBase, MmFvSize);
-  UsedSize += HobLength;
+  if (!IsMmPlatformFvHobCreated (PlatformHobList, PlatformHobSize)) {
```
Comment (niruiyu) [inline StandaloneMmPkg/Drivers/StandaloneMmIplPei/MmFoundationHob.c:940]: Comments like "Skip to report FV that contains MmCore when Platform reports FV"

Comment (hongbin123) [reply StandaloneMmPkg/Drivers/StandaloneMmIplPei/MmFoundationHob.c]: updated

**Review by niruiyu — APPROVED** [review]

**Pull request merged**